cmd/cli/certificate_rotate.go

```diff
@@ -0,0 +1,352 @@
+		return nil
+	}
+	err = r.updateCertificate(oldMrc.Name, v1alpha2.InactiveIntent)
+	if err != nil {
+		return fmt.Errorf("unable to update MeshRootCertificate [%s] with role [%s]: %s", oldMrc.Name, v1alpha2.InactiveIntent, err)
+	}
+	fmt.Fprintf(r.out, "waiting for propagation...\n\n")
+	time.Sleep(r.waitForRotation)
+
+	if r.deleteOld {
+		if !r.promptForContinue("Are you sure you want to delete the old MeshRootCertificate?") {
+			return nil
+		}
+		err = r.configClient.ConfigV1alpha2().MeshRootCertificates(settings.Namespace()).Delete(context.Background(), oldMrc.Name, metav1.DeleteOptions{})
+		if err != nil {
+			fmt.Printf("warning: unable to delete MeshRootCertificate [%s]", oldMrc.Name)
+		}
+		if oldMrc.Spec.Provider.Tresor != nil {
+			err = r.clientSet.CoreV1().Secrets(oldMrc.Spec.Provider.Tresor.CA.SecretRef.Namespace).Delete(context.Background(), oldMrc.Spec.Provider.Tresor.CA.SecretRef.Name, metav1.DeleteOptions{})
+			if err != nil {
+				fmt.Printf("warning: unable to delete secret [%s]", oldMrc.Spec.Provider.Tresor.CA.SecretRef.Name)
+			}
+		}
+	}
+
+	fmt.Fprintf(r.out, "\nOSM successfully rotated root certificate for mesh [%s] in namespace [%s]\n", r.meshName, settings.Namespace())
+	fmt.Fprintf(r.out, "\nThe new MeshRootCertificate [%s] is now active\n", newMrc.Name)
+	return nil
+}
+
+func (r *rotateCmd) findCurrentInUse() (*v1alpha2.MeshRootCertificate, error) {
+	mrcs, err := r.configClient.ConfigV1alpha2().MeshRootCertificates(settings.Namespace()).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("not able to find MeshRootCertificates: %w", err)
+	}
+
+	var activeMrc *v1alpha2.MeshRootCertificate
+	for _, mrc := range mrcs.Items {
+		if mrc.Spec.Intent == v1alpha2.ActiveIntent {
+			if activeMrc != nil {
+				return nil, fmt.Errorf("can only rotate when there is only one active MRC. Found two: %s and %s", mrc.Name, activeMrc.Name)
+			}
+			m := mrc
+			activeMrc = &m
+		}
+
+		if mrc.Spec.Intent == v1alpha2.PassiveIntent {
+			return nil, fmt.Errorf("it appears a rotation is in progress. \n\nFound MRC %s in state %s", mrc.Name, mrc.Spec.Intent)
+		}
+	}
+
+	if activeMrc == nil {
+		return nil, fmt.Errorf("not able to find MeshRootCertificate in active state")
+	}
+
+	return activeMrc, nil
+}
+
+func (r *rotateCmd) updateCertificate(name string, intent v1alpha2.MeshRootCertificateIntent) error {
+	mrc, err := r.configClient.ConfigV1alpha2().MeshRootCertificates(settings.Namespace()).Get(context.Background(), name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	mrc.Spec.Intent = intent
+	_, err = r.configClient.ConfigV1alpha2().MeshRootCertificates(settings.Namespace()).Update(context.Background(), mrc, metav1.UpdateOptions{})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *rotateCmd) createOrFindNewMRC(mrc *v1alpha2.MeshRootCertificate) (*v1alpha2.MeshRootCertificate, error) {
```

Comment:
nit: I'd consider having separate methods for create, createFromFile, and Find, and bring out the control logic outside of those methods.

This took me a sec to understand why it was structured this way.